homeassistant/components/statistics/sensor.py

```diff
@@ -203,7 +203,11 @@ def name(self):
     @property
     def native_value(self):
         """Return the state of the sensor."""
-        return self.mean if not self.is_binary else self.count
+        if self.is_binary:
+            return self.count
+        if self._precision == 0:
+            return int(self.mean)
```

Comment:
Don't we need to round instead of truncate?

Reply:
I don't think so. If the precision is 0, `mean` is already the rounded value with a precision of 0, isn't it?
But I see a different problem here. If the generation of mean/median fails they are set to STATE_UNKNOWN. 
https://github.com/home-assistant/core/blob/d98ea2aae8777d6b903372a470da2085e00e189a/homeassistant/components/statistics/sensor.py#L293
Trying to convert STATE_UNKNOWN to int will raise an ValueError. In addition mean is initialized with None, which can also not be converted to int. I would propose to check, if mean is a number

Reply:
Yes as @micha91 said, the value is already rounded.

Good catch on the second point! Why didn't I see this half a year ago 🤔 Fixed!

Reply:
@micha91 Bonus question: Is None even a state to be passed to the outside or should we convert it to STATE_UNKNOWN or similar? What does None represent? In another case I recently saw that it isn't well handled by the History function of HA for example.

Reply:
I think this would be a good way to go. I can't confirm that None is handled in a bad way, but in my opinion None is also an unknown state so it should be better to return STATE_UNKNOWN in this case

Reply:
@emontnemery your guidance?

Reply:
If an entity's state is set to `None`, Home Assistant converts it to the string "unknown" / `STATE_UNKNOWN`.

Reply:
You could do like this instead:
```py
        if self._precision == 0:
            with contextlib.suppress(TypeError)
                return int(self.mean)
        return self.mean
```

Reply:
Okay so no special handling of None. Fine.

Thanks, I've taken over your recommendation